### PR TITLE
Nikhil_Assignment_22_Done_(5+15=)20Min

### PR DESCRIPTION
**Pull request by NikhilMJun20**
->Completed and pushed Assignment # 22.
->Please review and comment.

->PFB screenshots of the outputs of the programs:-

**1**
Output_PalindromeNumber

**2**
Output_PalindromeString

### CODE REVIEW

File: src/nikhilMJun20/PalindromeString.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+package nikhilMJun20;
+
+import java.util.Scanner;
+
+public class PalindromeString {
+	void checkPalindromeString(String str) {
+		boolean flag = true;
+		str = str.toLowerCase();
+		for(int leftIndex=0, rightIndex=str.length()-1; leftIndex<=rightIndex; leftIndex++,rightIndex--) {
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/PalindromeString.java:17]: -- Your code and logic looks good to me.
--However I think you can simplify it a little.
--Refer below code:
String reverse="", originalName = name;
		for(int index=name.length()-1;index>=0;index--) {
			reverse = reverse+name.charAt(index);		
		}
		if(originalName.equals(reverse))
			System.out.println(" Given String " +originalName +" is Palindrome");
		else
			System.out.println(" Given String " +originalName +" is not Palindrome");

**Review by VaishnaviVJun20 — APPROVED** [review]
--Code looks good to me

File: src/nikhilMJun20/PalindromeString.java (hunk 1 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/PalindromeString.java:17]: ->Yes, it is a simple and string reverse logic. But, in my logic I do not required an additional string for keeping the reversed string and at the first mismatch - where one incrementing and one decrementing index pointer is used, I will be able to tell that the string is not a palindrome.
->Thanks for the review comment.

Comment (KrishnaKTechnocredits) [reply src/nikhilMJun20/PalindromeString.java:17]: @VaishnaviVJun20 Nikhil's logic is correct and as expected. Not required another string to store all characters in reverse order.

File: src/nikhilMJun20/PalindromeNumber.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+package nikhilMJun20;
+
+import java.util.Scanner;
+
+public class PalindromeNumber {
+	void checkPalindromeNumber(int number) {
+		int revNumber = 0, originalNumber=number;
+		while(number>0) {
+			revNumber = 10*revNumber+(number%10);
+			number = number/10;
```
Comment (KrishnaKTechnocredits) [inline src/nikhilMJun20/PalindromeNumber.java:18]: To reduce this line, you can write operation on while condition like below
while((number = number/10)>0)

**Review by KrishnaKTechnocredits — APPROVED** [review]
Code looks good to me.

File: src/nikhilMJun20/PalindromeNumber.java (hunk 2 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/PalindromeNumber.java:18]: Yes. This becomes even more compact.
Next time on I will use this style whenever possible.
Thank you, @KrishnaKTechnocredits !

**Pull request merged**